File: homework/src/main/java/com/mikheev/homework/controller/GenreController.java
package com.mikheev.homework.controller;

import com.mikheev.homework.controller.dto.GenreDto;
import com.mikheev.homework.domain.Genre;
import com.mikheev.homework.service.GenreService;
import lombok.RequiredArgsConstructor;
import org.modelmapper.ModelMapper;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.stream.Collectors;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/library")
public class GenreController {

    private final GenreService genreService;
    private final ModelMapper modelMapper;

    @GetMapping("/genre")
    public List<GenreDto> genreList() {
        return genreService.getAllGenres().stream()
                .map(genre -> modelMapper.map(genre, GenreDto.class)).collect(Collectors.toList());
    }
}

File: homework/src/main/kotlin/com/mikheev/homework/controller/GenreController.kt
package com.mikheev.homework.controller

import com.mikheev.homework.domain.Genre
import com.mikheev.homework.service.GenreService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/library")
class GenreController (val genreService: GenreService) {

    @GetMapping("/genre")
    fun genreList(): List<Genre> {
        return genreService.getAllGenres()
    }
}
